azext_iot/operations/events3/_events.py

```diff
@@ -127,13 +129,34 @@ def _output_msg_kpi(msg):
         event_source = {'event': {}}
 
         event_source['event']['origin'] = origin
-        event_source['event']['payload'] = str(next(msg.get_data()), 'utf8')
+
+        payload = ''
+
+        data = msg.get_data()
+        if data:
+            payload = str(next(data), 'utf8')
+
+        system_props = unicode_binary_map(parse_entity(msg.properties, True))
+
+        ct = content_type
+        if not ct:
+            ct = system_props['content_type'].lower() if 'content_type' in system_props else ''
```

Comment:
You could skip the `system_props['content_type'].lower()` here, then in the next line do `ct = ct.lower()` to lowercase once, content type either from explicit parameter or from msg payload.

Reply:
That's exactly what I just did :)